api-report/tree.api.md

```diff
@@ -4,9 +4,52 @@
 
 ```ts
 
+import { Serializable } from '@fluidframework/datastore-definitions';
+
+// @public (undocumented)
+export interface INodeReader {
+    // (undocumented)
+    keys: Iterable<TreeKey>;
+    // (undocumented)
+    length(key: TreeKey): number;
+    // (undocumented)
+    type: TreeType;
+    // (undocumented)
+    value: undefined | Serializable;
+}
+
+// @public (undocumented)
+export interface ITreeReader extends INodeReader {
+    // (undocumented)
+    pop(): TreeReadResult;
+    // (undocumented)
+    push(key: TreeKey, index: number): TreeReadResult;
+}
+
 // @public (undocumented)
-export const _ = "Force index.d.ts generation from ApiExtractor.";
+export type TreeId = number | string & {
+    readonly TreeId: symbol;
+};
 
+// @public (undocumented)
+export type TreeKey = number | string & {
+    readonly TreeKey: symbol;
+};
+
+// @public (undocumented)
+export const enum TreeReadResult {
+    // (undocumented)
+    NotFound = -1,
+    // (undocumented)
+    Ok = 1,
+    // (undocumented)
+    Pending = 0
```

Comment:
What does `Pending` mean? Is this pending due to lazy loading, or just pending for the cursor? Perhaps an example of usage of this system would help me understand.

Reply:
Pending means navigation failed due to tree chunks being unavailable.  I added a doc comment for this.

(Everything will get doc comments soon.  I'm just waiting for naming to quiesce.)